docs/development/virtual_environments.rst

```diff
@@ -129,10 +129,126 @@ virtualized, rather than running on hardware. You will need to
+
+Switch Vagrant provider to libvirt - Debian 9
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                                                                                
+Install Vagrant, libvirt, QEMU, and their dependencies:
+                                                                                
+.. code:: sh
+
+   sudo apt-get update
+   sudo apt-get install -y vagrant vagrant-libvirt libvirt-daemon-system qemu-kvm
+   sudo apt-get install -y ansible rsync
+   vagrant plugin install vagrant-libvirt
+   sudo usermod -a -G libvirt $USER 
+   sudo systemctl restart libvirtd                                                                    
+                                                                                
+Add your user to the kvm group to give it permission to run KVM:                                            
+                                                                                
+.. code:: sh
+
+   sudo usermod -a -G kvm $USER
+   sudo rmmod kvm_intel
+   sudo rmmod kvm
+   sudo modprobe kvm
+   sudo modprobe kvm_intel
+                                                                    
+                                                                                
+Log out, then log in again. Verify that libvirt is installed and KVM is         
+available:                                                                      
+                                                                                
+.. code:: sh                                                                    
+                                                                                
+   libvirtd --version                                                           
+   kvm-ok                                                                       
+                                                                                
+Convert the bento/14.04 Vagrant box from ``virtualbox`` to ``libvirt`` format:  
+                                                                                
+.. code:: sh                                                                    
+                                              
+   sudo apt-get install -y vagrant-mutate                                  
+   vagrant box add --provider virtualbox bento/ubuntu-14.04                     
+   vagrant mutate bento/ubuntu-14.04 libvirt                                    
+                                                                                
+Set the default Vagrant provider to ``libvirt``:                                
+                                                                                
+.. code:: sh                                                                    
+                                                                                
+   echo 'export VAGRANT_DEFAULT_PROVIDER=libvirt' >> ~/.bashrc                  
+   export VAGRANT_DEFAULT_PROVIDER=libvirt                                      
```

Comment:
As with the mutate steps, the bashrc customizations need not be restated for both Debian and Ubuntu—once is enough, if set clearly.